fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/temp/DotExporter.java

```diff
@@ -138,22 +140,93 @@ protected void renderGraphAttributes(final PrintWriter out, final Map<String, St
     }
 
     @Override
-    protected void renderNode(final PrintWriter out, final N node, final Map<String, String> attributes) {
-        out.print(INDENT);
-        out.print(getVertexID(node));
+    protected void renderNodes(final ExporterContext context) {
+        final ImmutableNetwork<N, E> network = context.getNetwork();
+        final PrintWriter out = context.getPrintWriter();
+
+        super.renderNodes(context);
 
-        renderAttributes(out, attributes);
+        // Go through the vertex set a second time to render dependsOn information
+        // vertex set
+        for (final N n : network.nodes()) {
+            // If the current node had children that explicitly encode an order we do the following:
+            // We create a sub-block and set rank=same, rankDir=LR in order to have all children be rendered
+            // on the same level but left to right -- this is especially important for showing join-orders
+            // left-to-right.
+
+            final Set<E> childrenEdges = network.inEdges(n);
+
+            // We sort the childrenEdges topologically insertion sort-style O(N^2).
+            final List<E> orderedChildrenEdges = new ArrayList<>(childrenEdges.size());
+
+            boolean needsInvisibleEdges = false;
+            for (final E toBeInsertedEdge : childrenEdges) {
+                final Set<? extends PlannerGraph.AbstractEdge> dependsOn = toBeInsertedEdge.getDependsOn();
+
+                if (!dependsOn.isEmpty()) {
+                    needsInvisibleEdges = true;
+                }
+
+                int index = 0;
+                for (; index < orderedChildrenEdges.size(); index ++) {
+                    final E currentEdge = orderedChildrenEdges.get(index);
+                    if (!dependsOn.contains(currentEdge)) {
+                        break;
+                    }
+                }
```

Comment:
If we keep this general block of code, I think this is more in line with our style guide:
```suggestion
                while (index < orderedChildrenEdges.size()) {
                    final E currentEdge = orderedChildrenEdges.get(index);
                    if (!dependsOn.contains(currentEdge)) {
                        break;
                    }
                    index++;
                }
```

Reply:
Done.